apps/vmq_server/src/vmq_mqtt_fsm.erl

```diff
@@ -1034,6 +1042,10 @@ dispatch_publish_qos0(_MessageId, Msg, State) ->
             %% we have to close connection for 3.1.1
             _ = vmq_metrics:incr_mqtt_error_auth_publish(),
             {error, not_allowed};
+        {error, not_allowed} when DisconnectOnUnauthorizedPublishV3 ->
```

Comment:
Note we could use a single clause instead of 2, using a guard sequence:
`{error, not_allowed} when ?IS_PROTO_4(Proto) ; DisconnectOnUnauthorizedPublishV3 ->`   ... etc